### ISSUE TITLE
[REVIEW]: ROSS - Rotordynamic Open Source Software

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2120]
**Submitting author:** @raphaeltimbo (<a href="http://orcid.org/0000-0001-7689-5486">Raphael Silva</a>)
**Repository:** <a href="https://github.com/ross-rotordynamics/ross" target ="_blank">https://github.com/ross-rotordynamics/ross</a>
**Version:** v0.3.2
**Editor:** @meg-simula
**Reviewer:** @srmnitc, @nnadeau
**Archive:** <a href="https://doi.org/10.5281/zenodo.3746582" target="_blank">10.5281/zenodo.3746582</a>

## Status

[![status](https://joss.theoj.org/papers/7006399ebbe19bfb187a194789c862d7/status.svg)](https://joss.theoj.org/papers/7006399ebbe19bfb187a194789c862d7)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/7006399ebbe19bfb187a194789c862d7"><img src="https://joss.theoj.org/papers/7006399ebbe19bfb187a194789c862d7/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/7006399ebbe19bfb187a194789c862d7/status.svg)](https://joss.theoj.org/papers/7006399ebbe19bfb187a194789c862d7)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/ross-rotordynamics/ross" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@srmnitc & @nnadeau, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @meg-simula know.

✨ **Please try and complete your review in the next two weeks** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/ross-rotordynamics/ross">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@raphaeltimbo) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/ross-rotordynamics/ross">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@raphaeltimbo) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [ ] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @srmnitc, @nnadeau it looks like you're currently assigned to review this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1017/CBO9780511780509 is OK
- 10.1109/MCSE.2011.37 is OK
- 10.1002/9780470903704 is OK
- 10.1002/9783527651894 is OK
- 10.1007/978-3-662-09786-1 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02120/joss.02120/10.21105.joss.02120.pdf)

**Comment by meg-simula** [comment]
@nnadeau @srmnitc  Thanks again for agreeing to review. Reviewer checklists have been generated for each you above - please feel free to dig in at will!

**Comment by meg-simula** [comment]
@srmnitc @nnadeau @raphaeltimbo :wave: Just checking in - how are matters progressing?

**Comment by srmnitc** [comment]
@meg-simula I took a first look, everything seems to be well documented and easy to review. I hope to finish my first round of review in 2-3 days !

**Comment by srmnitc** [comment]
@meg-simula @raphaeltimbo I have completed my first round of review. I think the submission already meets most of the review checkpoints and is quite nicely documented with a lot of examples. I only have three main issues - with installation, tests and citations in the paper. I have raised these issues on the repository. The jupyter notebooks are quite helpful, I will also go a bit more into detail in the next one or two days.

**Comment by arfon** [comment]
Dear authors and reviewers

We wanted to notify you that in light of the current COVID-19 pandemic, JOSS has decided to suspend submission of new manuscripts and to handle existing manuscripts (such as this one) on a "best efforts basis". We understand that you may need to attend to more pressing issues than completing a review or updating a repository in response to a review. If this is the case, a quick note indicating that you need to put a "pause" on your involvement with a review would be appreciated but is not required.

Thanks in advance for your understanding.

_Arfon Smith, Editor in Chief, on behalf of the JOSS editorial team._

**Comment by srmnitc** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02120/joss.02120/10.21105.joss.02120.pdf)

**Comment by srmnitc** [comment]
@meg-simula @raphaeltimbo All the issues I have raised were cleared.  I am quite happy with the state of the code and wholeheartedly recommend the paper for publication. Thanks @raphaeltimbo for the nice module!

**Comment by arfon** [comment]
:wave: @nnadeau - how are you getting on with your review?

**Comment by nnadeau** [comment]
@arfon in progress, waiting on:

- https://github.com/ross-rotordynamics/ross/issues/495
- https://github.com/ross-rotordynamics/ross/issues/494
- https://github.com/ross-rotordynamics/ross/issues/493
- https://github.com/ross-rotordynamics/ross/issues/492

**Comment by nnadeau** [comment]
@raphaeltimbo my checklist is complete and I'm just waiting to resolve the issues listed below. Once those are resolved, we're good to go! :) 

https://github.com/ross-rotordynamics/ross/issues/492
https://github.com/ross-rotordynamics/ross/issues/494
https://github.com/ross-rotordynamics/ross/issues/495
https://github.com/ross-rotordynamics/ross/issues/484
https://github.com/ross-rotordynamics/ross/issues/480
https://github.com/ross-rotordynamics/ross/pull/516
https://github.com/ross-rotordynamics/ross/issues/482

**Comment by nnadeau** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02120/joss.02120/10.21105.joss.02120.pdf)

**Comment by nnadeau** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1017/CBO9780511780509 is OK
- 10.1109/MCSE.2011.37 is OK
- 10.1115/1.1349417 is OK
- 10.1007/s40430-019-1652-8 is OK

MISSING DOIs

- None

INVALID DOIs

- https://doi.org/10.1038/s41592-019-0686-2 is INVALID because of 'https://doi.org/' prefix
```

**Comment by nnadeau** [comment]
@arfon @meg-simula @raphaeltimbo we're good to go! good work!

@raphaeltimbo i patched the DOI issue here: https://github.com/ross-rotordynamics/ross/pull/518

**Comment by arfon** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1017/CBO9780511780509 is OK
- 10.1109/MCSE.2011.37 is OK
- 10.1038/s41592-019-0686-2 is OK
- 10.1115/1.1349417 is OK
- 10.1007/s40430-019-1652-8 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by arfon** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02120/joss.02120/10.21105.joss.02120.pdf)

**Comment by raphaeltimbo** [comment]
> **Submitting author:** @raphaeltimbo ([Raphael Silva](http://orcid.org/0000-0001-7689-5486))
> **Repository:** https://github.com/ross-rotordynamics/ross
> **Version:** v0.2.0
> **Editor:** @meg-simula
> **Reviewer:** @srmnitc, @nnadeau
> **Archive:** Pending

@meg-simula, with the review process we had to make some modifications to the package and the final version now is v0.3.2. Can we update that information?

**Comment by arfon** [comment]
:wave: @raphaeltimbo - I'll take over this paper from here as I'm not sure about @meg-simula's availability given the COVID-19 stuff going on.

I've read your paper and have a few suggested changes to improve the readability in https://github.com/ross-rotordynamics/ross/pull/519 .

**Comment by arfon** [comment]
@whedon set v0.3.2 as version

**Comment by whedon** [comment]
OK. v0.3.2 is the version.

**Comment by arfon** [comment]
@raphaeltimbo - At this point could you make a new release of this software that includes the changes that have resulted from this review. Then, please make an archive of the software in Zenodo/figshare/other service and update this thread with the DOI of the archive? For the Zenodo/figshare archive, please make sure that:

- The title of the archive is the same as the JOSS paper title
- That the authors of the archive are the same as the JOSS paper authors

I can then move forward with accepting the submission.

**Comment by raphaeltimbo** [comment]
> @raphaeltimbo - At this point could you make a new release of this software that includes the changes that have resulted from this review. Then, please make an archive of the software in Zenodo/figshare/other service and update this thread with the DOI of the archive? For the Zenodo/figshare archive, please make sure that:
> 
> * The title of the archive is the same as the JOSS paper title
> * That the authors of the archive are the same as the JOSS paper authors
> 
> I can then move forward with accepting the submission.

@arfon, I have added the software to zenodo: http://doi.org/10.5281/zenodo.3746582

**Comment by arfon** [comment]
@whedon set 10.5281/zenodo.3746582 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.3746582" target="_blank">10.5281/zenodo.3746582</a> is the archive.

**Comment by arfon** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1017/CBO9780511780509 is OK
- 10.1109/MCSE.2011.37 is OK
- 10.1038/s41592-019-0686-2 is OK
- 10.1115/1.1349417 is OK
- 10.1007/s40430-019-1652-8 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1412

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1412, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by arfon** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1248325450457321473) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/1413
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.02120](https://doi.org/10.21105/joss.02120)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? notify your editorial technical team...

**Comment by arfon** [comment]
@srmnitc, @nnadeau - many thanks for your reviews here and to @meg-simula for editing this submission ✨ 

@raphaeltimbo - your paper is now accepted into JOSS :zap::rocket::boom:

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.02120/status.svg)](https://doi.org/10.21105/joss.02120)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.02120">
  <img src="https://joss.theoj.org/papers/10.21105/joss.02120/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.02120/status.svg
   :target: https://doi.org/10.21105/joss.02120
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.02120/status.svg)](https://doi.org/10.21105/joss.02120)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss

**Comment by raphaeltimbo** [comment]
Thanks @arfon and @meg-simula!
And thank you @srmnitc and @nnadeau for the review. It really helped us to improve the paper and the software!
